internal/worker/server.go

```diff
@@ -562,6 +560,27 @@ func (s *Server) FinishJob(token uuid.UUID, result json.RawMessage) error {
 	return nil
 }
 
+func getMetricsJobType(jobType string) string {
+	if strings.HasPrefix(jobType, JobTypeOSBuild+":") {
+		return JobTypeOSBuild
+	}
+	return jobType
+}
```

Comment:
From what I checked, we should be able to change the `Server.jobStatus()` to strip the osbuild job type in the same way as it is done by `Server.JobType()`. Especially since you won't need it for Prometheus any more.

IMO it will also make the code more sane and consistent, because the returned job type will be always the same, regardless of the used method.

The only place where this won't help is the method that handles enqueueing of jobs, where you'll still need to strip the architecture from job type. But there you could use `strings.Split(jobType, ":")[0]` instead as used by `JobType()`.